### PR TITLE
[sca/WIP] Update the ecdsa384_sca code

### PR DESCRIPTION
**Pull request by bilgiday**
Update ecc384_serial.c and p384_ecdsa_sca.s files to be compatible with the latest capture.py file in the ot-sca repo.

This is an WIP PR.

### CODE REVIEW

**Comment by bilgiday** [comment]
@wettermo, @jadephilipoom,

I modified these two files based on the `p256_serial.c` file in th OT-repo, and the `p384_ecdsa_sign_test.s` file in the PR #19290. 

I was able to compile the files using #19290. @wettermo, you can use them in your local environment and test if they work with the ot-sca tooling.

**Comment by wettermo** [comment]
Thanks a lot @bilgiday ! I'll keep you updated on my progress :) .

File: sw/device/sca/ecc384_serial.c
Comment (wettermo) [inline sw/device/sca/ecc384_serial.c:20]: Nit: *P384

**Review by wettermo — APPROVED** [review]
LGTM! I tested the changes on my PC/FPGA setup: no errors and compatible with capture.py in ot-sca repository.
Thanks a lot @bilgiday !!

**Comment by bilgiday** [comment]
@wettermo Thanks for checking it!

@jadephilipoom, It looks like the CI checks also pass now. If it looks okay to you, we can merge this. WDYT?

**Review by vrozic — APPROVED** [review]

**Comment by vrozic** [comment]
Thanks @bilgiday for working on this. The code looks good to me.

We can update the binaries in ot-sca once this is merged. For now, I've opened [ot-sca Issue 149](https://github.com/lowRISC/ot-sca/issues/149) so we don't forget.

**Comment by bilgiday** [comment]
@vrozic Great, thanks for opening the ot-sca-issue. I will update the binaries after this PR is merged.

**Comment by vogelpi** [comment]
This PR is a bit older than a month an in attempt to move things forward, I've taken the liberty to rebase this on current master.

**Comment by vogelpi** [comment]
Update: I've now also included the change suggested by @wettermo and fixed some more comments referring to P256 as well as a missing factor of two (was missing already before). I will now approve this PR and merge it once CI passes.

**Review by vogelpi — APPROVED** [review]
LGTM, thanks for updating these apps @bilgiday !

File: sw/device/sca/ecc384_serial.c
Comment (vogelpi) [reply sw/device/sca/ecc384_serial.c]: Fixed

**Review by jadephilipoom — APPROVED** [review]
Belated LGTM from me :slightly_smiling_face:

**Pull request merged**
